### PR TITLE
Fixed #25882 -- Prevented fast deletes matching no rows from crashing on MySQL.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

**Comment by charettes** [comment]
Commits are meant to be squashed. I only pushed them this way to exposed the failure on CI's MySQL without the fix.

**Comment by MulberryComic20** [comment]
Hello

File: tests/delete/tests.py [hunk 1]
```diff
@@ -503,3 +503,12 @@ def test_fast_delete_large_batch(self):
         # that + fast delete of the related objs.
         self.assertNumQueries(2, a.delete)
         self.assertEqual(User.objects.count(), 0)
+
+    def test_fast_delete_empty_no_update_can_self_select(self):
+        """
+        #25932 - Fast deleting on backends that don't have the
+        `no_update_can_self_select` feature should work event if the specified
```
Comment (sir-sigurd) [inline tests/delete/tests.py]: `even` instead of `event`?

File: docs/releases/1.9.1.txt [hunk 2]
```diff
@@ -34,3 +34,6 @@ Bugfixes
   created by ``startapp`` on Python 2 (:ticket:`25909`). Add this line to your
   own ``apps.py`` files created using Django 1.9 if you want your migrations
   to work on both Python 2 and Python 3.
+
+* Prevented some queryset involving ``JOIN`` to crash on ``delete()`` on MySQL
```
Comment (timgraham) [inline docs/releases/1.9.1.txt]: suggested: Prevented QuerySet.delete() from crashing on MySQL when querying across relations

File: tests/delete/tests.py (hunk 1 above)
Comment (timgraham) [inline tests/delete/tests.py]: Should we also add an assertion to demonstrate that the query doesn't match any row?

Comment (charettes) [inline tests/delete/tests.py]: @sir-sigurd right thanks! @timgraham wouldn't hurt.

File: docs/releases/1.9.1.txt (hunk 2 above)
Comment (charettes) [inline docs/releases/1.9.1.txt]: Will change.

**Pull request merged**